dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/relationship/hibernate/RelationshipStoreTest.java

```diff
@@ -102,85 +102,189 @@ public void setUpTest()
+
+        ProgramStageInstance programStageInstance = addProgramStageInstance( programInstance, programStageA );
+
+        trackedEntityInstanceA = createTrackedEntityInstance( organisationUnit );
+        trackedEntityInstanceService.addTrackedEntityInstance( trackedEntityInstanceA );
+
+        Relationship relationshipA = addTeiToProgramStageInstanceRelationShip( trackedEntityInstanceA,
+            programStageInstance );
+
         List<Relationship> relationshipList = relationshipService
             .getRelationshipsByProgramStageInstance( programStageInstance, true );
+
         assertEquals( 1, relationshipList.size() );
         assertTrue( relationshipList.contains( relationshipA ) );
+
         assertTrue( relationshipService.getRelationshipByRelationship( relationshipA ).isPresent() );
     }
 
     @Test
-    void getByRelationshipType()
+    void testGetByProgramInstance()
     {
+        trackedEntityInstanceA = createTrackedEntityInstance( organisationUnit );
+        trackedEntityInstanceService.addTrackedEntityInstance( trackedEntityInstanceA );
+
+        Program programA = addProgram();
+
+        ProgramInstance programInstance = addProgramInstance( programA );
+
+        Relationship relationshipA = addTeiToProgramInstanceRelationShip( trackedEntityInstanceA,
+            programInstance );
+
+        List<Relationship> relationshipList = relationshipService
+            .getRelationshipsByProgramInstance( programInstance, true );
+
+        assertEquals( 1, relationshipList.size() );
+        assertTrue( relationshipList.contains( relationshipA ) );
+
+        assertTrue( relationshipService.getRelationshipByRelationship( relationshipA ).isPresent() );
+    }
+
+    @Test
+    void testGetByRelationshipType()
+    {
+        addTeiToTeiRelationShip();
+
         List<Relationship> relationshipList = relationshipService
             .getRelationshipsByRelationshipType( relationshipType );
+
         assertEquals( 1, relationshipList.size() );
-        assertTrue( relationshipList.contains( relationship ) );
+        assertTrue( relationshipList.contains( teiRelationship ) );
     }
 
     @Test
     void testGetByRelationship()
     {
-        Optional<Relationship> existing = relationshipService.getRelationshipByRelationship( relationship );
+        addTeiToTeiRelationShip();
+
+        Optional<Relationship> existing = relationshipService.getRelationshipByRelationship( teiRelationship );
+
         assertTrue( existing.isPresent() );
     }
-}
+
+    void addTeiToTeiRelationShip()
+    {
+        trackedEntityInstanceA = createTrackedEntityInstance( organisationUnit );
+        trackedEntityInstanceB = createTrackedEntityInstance( organisationUnit );
+
+        trackedEntityInstanceService.addTrackedEntityInstance( trackedEntityInstanceA );
+        trackedEntityInstanceService.addTrackedEntityInstance( trackedEntityInstanceB );
+
+        teiRelationship = new Relationship();
```

Comment:
Can we just call this method `relationship()` and let it return a `Relationship`. I personally find these helpers that are not using some fields easier to read/reason about.